### DISCUSSION TITLE
How to use Multiple criteria

### DISCUSSION THREAD

**Discussion by kyky19831214** [fastapi/fastapi#5592] [Questions] [answered]
I want to use "session.query()" get a list object from mysql table and there are 10 criteria.
I don`t know how to code with one sentence.
for example:
session.query(User). filter(User.name == 'aa')
session.query(User). filter(User.name == 'aa', User.age > 5)
session.query(User). filter(User.name == 'aa', User.age > 5, sex=='boy')
....

need help !

**Comment by iudeen** [comment, marked as answer]
Stackoverflow might be a better place to ask general questions like this! 

Given that, if you're looking a starting point, check [this](https://stackoverflow.com/a/44702268)!

**Reply by kyky19831214** [reply]
thank you for your advise.
and I try to use "eval()" function, like this :

    criteria = ""
    if not condition.vin is None:
        criteria = criteria + "models.TAccidentItem.vin == condition.vin" + ","
    if not condition.car_type is None:
        criteria = criteria + "models.TAccidentItem.car_type == condition.car_type" + ","
    if not condition.accident_time_start is None:
        criteria = criteria + "models.TAccidentItem.accident_time >= condition.accident_time_start" + ","
    if not condition.accident_time_end is None:
        criteria = criteria + "models.TAccidentItem.accident_time<= condition.accident_time_end" + ","
    if not condition.accident_level is None:
        criteria = criteria + "models.TAccidentItem.accident_level == condition.accident_level" + ","
    if not condition.ota_version is None:
        criteria = criteria + "models.TAccidentItem.ota_version == condition.ota_version" + ","
    if not condition.adas_level is None:
        if condition.adas_level == '1':
            criteria = criteria + "models.TAccidentItem.adas_status == 1" + ","
        elif condition.adas_level == '2':
            criteria = criteria + "models.TAccidentItem.lka_status == 1" + ","
        elif condition.adas_level == '3':
            criteria = criteria + "models.TAccidentItem.noa_status == 1" + ","
        elif condition.adas_level == '4':
            criteria = criteria + "models.TAccidentItem.auto_status == 2" + ","

    criteria = criteria + "models.TAccidentItem.status in (2)"
    all_criteria = "query_rs.filter(" + criteria + ")"

    print(all_criteria)
    dd = eval(all_criteria)
    dd = dd.first()

it works.

**Reply by tiangolo** [reply]
Thanks @iudeen! 🚀

Yep, indeed, as this is not related to FastAPI it might be better on StackOverflow.
